libs/feature/search/src/lib/favorites/favorite-star/favorite-star.component.spec.ts

```diff
@@ -115,54 +115,90 @@ describe('FavoriteStarComponent', () => {
       describe('if record is not part of favorite', () => {
         beforeEach(() => {
           ;(favoritesService as any).myFavoritesUuid$.next([
-            'aaa',
-            'bbb',
-            'ccc',
+            component.record.uuid,
           ])
-          starToggle.newValue.emit(true)
           fixture.detectChanges()
         })
-        it('adds record to favorites', () => {
-          expect(favoritesService.addToFavorites).toHaveBeenCalledWith([
-            component.record.uuid,
-          ])
-          expect(favoritesService.removeFromFavorites).not.toHaveBeenCalled()
+        describe('if change is made from this component', () => {
+          beforeEach(() => {
+            starToggle.newValue.emit(true)
+            fixture.detectChanges()
+          })
+          it('adds record to favorites', () => {
+            expect(favoritesService.addToFavorites).toHaveBeenCalledWith([
+              component.record.uuid,
+            ])
+            expect(favoritesService.removeFromFavorites).not.toHaveBeenCalled()
+          })
+          it('increase record favorite count by one', () => {
```

Comment:
Damn I forgot to commit one comment

While this test is a consequence of `myFavoritesUuid$` tick.
Both things are decorrelated and should be tested separatly.

When I click on the star => here you test that the functions have been called
  - When the record is already in favorite
  - When the record is not in favorites
When favorites has been updated => here you test that the amount is correct
  - My record have been updated
  - My record haven't been updated